includes/sanitizers/class-amp-comments-sanitizer.php

```diff
@@ -62,130 +80,196 @@ public function sanitize() {
+			$reply_heading_element->replaceChild( $reply_heading_text_span, $reply_heading_text_node );
+			$reply_heading_text_span->appendChild( $reply_heading_text_node );
+
+			// Move whitespace after the node.
+			$reply_heading_text_node->nodeValue = rtrim( $reply_heading_text_node->nodeValue );
+			$reply_heading_element->insertBefore(
+				$this->dom->createTextNode( ' ' ),
+				$reply_heading_text_span->nextSibling
+			);
+
+			// Note: if the replytocom query parameter was set, then the existing value will already be a replyTo value.
+			// Nevertheless, the link will have the replytocom arg removed, so clicking on the link will cause
+			// navigation to a page that has the nameless heading text.
+			$text_binding = sprintf(
+				'%1$s.replyTo ? %1$s.replyTo : %2$s',
+				$state_id,
+				wp_json_encode( $reply_heading_text_node->nodeValue, JSON_UNESCAPED_UNICODE )
+			);
+
+			$reply_heading_text_span->setAttribute(
+				Amp::BIND_DATA_ATTR_PREFIX . 'text',
+				$text_binding
+			);
+		}
+
+		// Update comment reply links to set the reply state.
+		$comment_reply_links = $this->dom->xpath->query( '//a[ @data-commentid and @data-postid and @data-replyto and @data-respondelement and contains( @class, "comment-reply-link" ) ]' );
+		foreach ( $comment_reply_links as $comment_reply_link ) {
+			/** @var Element $comment_reply_link */
+
+			$comment_reply_state = [
+				$state_id => [
+					'replyTo'       => $comment_reply_link->getAttribute( 'data-replyto' ),
+					'commentParent' => $comment_reply_link->getAttribute( 'data-commentid' ),
+				],
+			];
+
+			$comment_reply_link->setAttribute(
+				Attribute::HREF,
+				'#' . $comment_reply_link->getAttribute( 'data-respondelement' )
+			);
+
+			$comment_reply_link->addAmpAction(
+				'tap',
+				sprintf(
+					'AMP.setState(%s),comment.focus',
+					wp_json_encode( $comment_reply_state, JSON_UNESCAPED_UNICODE )
+				)
+			);
+		}
+
+		$cancel_comment_reply_link = $this->dom->getElementById( 'cancel-comment-reply-link' );
+		if ( $cancel_comment_reply_link instanceof Element ) {
+
+			// Use hidden attribute to hide/show cancel reply link when commentParent is zero.
+			$cancel_comment_reply_link->removeAttribute( Attribute::STYLE );
+			if ( ! $comment_parent_id ) {
+				$cancel_comment_reply_link->setAttributeNode( $this->dom->createAttribute( Attribute::HIDDEN ) );
+			}
+			$cancel_comment_reply_link->setAttribute(
+				Amp::BIND_DATA_ATTR_PREFIX . Attribute::HIDDEN,
+				sprintf( '%s.commentParent == "0"', $state_id )
+			);
+
+			// Reset state when clicking cancel.
+			$cancel_comment_reply_link->addAmpAction(
+				'tap',
+				sprintf( 'AMP.setState({%s: %s})', $state_id, wp_json_encode( $state, JSON_UNESCAPED_UNICODE ) )
+			);
+		}
+	}
+
+	/**
+	 * Prepare for native comment-reply functionality.
+	 *
+	 * @param Element $comment_reply_script Comment reply script.
+	 */
+	protected function prepare_native_comment_reply( Element $comment_reply_script ) {
```

Comment:
@pierlon Any idea why there is reported coverage missing for this method? 

![image](https://user-images.githubusercontent.com/134745/135370358-ace7f462-bd70-4aee-b4a2-175901eda69f.png)

I can confirm that all the code is executed as part of `Test_AMP_Comments_Sanitizer::test_ampify_threaded_comments()`. I guess I should add `@covers ::prepare_native_comment_reply()` to that test? I didn't think that was necessary since this is not a public method.

Reply:
Seems those results were stale and it's now being [reported as covered](https://app.codecov.io/gh/ampproject/amp-wp/compare/6546/changes#D6L251) by Codecov:

![image](https://user-images.githubusercontent.com/16200219/135406849-ef4d41f2-214d-46ef-a86b-a95d01f371aa.png)

Reply:
I think that's because I explicitly added the `@covers` tag in 4e5e4fd68428a8ca0863ba5f573d94a9a0255ab7. But it seems unnecessary since the method is not public.